packages/uui-text-copy/lib/uui-text-copy.element.ts

```diff
@@ -0,0 +1,171 @@
+import { demandCustomElement } from '@umbraco-ui/uui-base/lib/utils';
+import { UUIButtonElement } from '@umbraco-ui/uui-button/lib';
+import { UUIInterfaceColor, UUIInterfaceLook } from '@umbraco-ui/uui-base';
+import { LabelMixin } from '@umbraco-ui/uui-base/lib/mixins';
+import { UUITextCopyEvent } from './UUITextCopyEvent';
+
+/**
+ * @summary A button to trigger text content to be copied to the clipboard
+ * @element uui-text-copy
+ * @dependency uui-button
+ * @dependency uui-icon
+ * @fires {UUITextCopyEvent} copying - Fires before the content is about to copied to the clipboard and can be used to transform or modify the data before its added to the clipboard
+ * @fires {UUITextCopyEvent} copied - Fires when the content is copied to the clipboard
+ * @slot - Use to replace the default content of 'Copy' and the copy icon
+ */
+@defineElement('uui-text-copy')
+export class UUITextCopyElement extends LabelMixin('', LitElement) {
+  /**
+   * Set a string you wish to copy to the clipboard
+   * @type {string}
+   * @default ''
+   */
+  @property({ type: String })
+  value = '';
+
+  /**
+   * Disables the button
+   * @type {boolean}
+   * @attr
+   * @default false
+   */
+  @property({ type: Boolean })
+  disabled = false;
+
+  /**
+   * Copies the text content from another element by specifying the ID of the element
+   * The ID of the element does not need to start with # like a CSS selector
+   * If this property is set, the value property is ignored
+   * @type {string}
+   * @attr
+   * @default ''
+   * @example copy-from="element-id"
+   */
+  @property({ type: String, attribute: 'copy-from' })
+  copyFrom = '';
+
+  /**
+   * Changes the look of the button to one of the predefined, symbolic looks.
+   * @type {"default" | "primary" | "secondary" | "outline" | "placeholder"}
+   * @attr
+   * @default "default"
+   */
+  @property()
+  look: UUIInterfaceLook = 'default';
+
+  /**
+   * Changes the look of the button to one of the predefined, symbolic looks. For example - set this to positive if you want nice, green "confirm" button.
+   * @type {"default" | "positive" | "warning" | "danger"}
+   * @attr
+   * @default "default"
+   */
+  @property({ reflect: true })
+  color: UUIInterfaceColor = 'default';
+
+  /**
+   * Makes the left and right padding of the button narrower.
+   * @type {boolean}
+   * @attr
+   * @default false
+   */
+  @property({ type: Boolean })
+  compact = false;
+
+  /**
+   * The delay in milliseconds before the button returns to its default state after a successful copy
+   * @type {number}
+   * @attr
+   * @default 250
+   */
+  @property({ type: Number, attribute: 'animation-state-delay' })
+  animationStateDelay = 250;
+
+  constructor() {
+    super();
+    demandCustomElement(this, 'uui-button');
+    demandCustomElement(this, 'uui-icon');
+  }
+
+  // Used to store the value that will be copied to the clipboard
+  #valueToCopy = '';
```

Comment:
I dont see #valueToCopy to be a class property, it could be a `let` inside the function below.